### PR TITLE
[Spark][Test-Only] Add Row Tracking Clone tests with empty source table

### PR DESCRIPTION
**Pull request by longvu-db**
#### Which Delta project/connector is this regarding?

- [X] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
Add Materialized Row Tracking columns CLONE tests, making sure we have enough Row Tracking test coverage necessary to gear us towards [enabling Row Tracking outside of testing](https://github.com/delta-io/delta/pull/2059).

## How was this patch tested?
Existing + added UTs.

## Does this PR introduce _any_ user-facing changes?
No.

### CODE REVIEW

File: spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala [hunk 1]
```diff
@@ -70,6 +70,46 @@ class RowIdCloneSuite
     }
   }
 
+  test("clone from empty source into an empty or an non-existing target " +
```
Comment (longvu-db) [inline spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala:73]: Will remove "an" in "an non-existing target".

File: spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala [hunk 2]
```diff
@@ -70,6 +70,48 @@ class RowIdCloneSuite
     }
   }
 
+  for (rowIdsEnabledOnSource <- BOOLEAN_DOMAIN) {
+    for (targetTableState <- Seq(TableState.EMPTY, TableState.NON_EXISTING))
```
Comment (fred-db) [inline spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala:74]:
```suggestion
  for {
    rowIdsEnabledOnSource <- BOOLEAN_DOMAIN
    targetTableState <- Seq(TableState.EMPTY, TableState.NON_EXISTING)
  } {
```

File: spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala [hunk 3]
```diff
@@ -70,6 +70,48 @@ class RowIdCloneSuite
...
+      s"targetTableState=$targetTableState") {
+      withTables(
+        TableSetupInfo(tableName = "source",
+          rowIdsEnabled = rowIdsEnabledOnSource, tableState = TableState.EMPTY),
+        TableSetupInfo(tableName = "target",
+          rowIdsEnabled = false, tableState = targetTableState)) {
+        cloneTable(targetTableName = "target", sourceTableName = "source")
+
+        val (targetLog, snapshot) =
+          DeltaLog.forTableWithSnapshot(spark, TableIdentifier("target"))
```
Comment (fred-db) [inline spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala:86]:
```suggestion
        val (targetLog, snapshot) = DeltaLog.forTableWithSnapshot(spark, TableIdentifier("target"))
```

File: spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala [hunk 4]
```diff
@@ -70,6 +70,48 @@ class RowIdCloneSuite
...
+        assertRowIdsAreNotSet(targetLog)
+        assert(RowId.isSupported(snapshot.protocol) === rowIdsEnabledOnSource)
+        assert(RowId.isEnabled(snapshot.protocol, snapshot.metadata) === rowIdsEnabledOnSource)
+      }
+    }
+  }
+
+  for (targetTableState <- Seq(TableState.EMPTY, TableState.NON_EXISTING))
+  test("clone from empty source on an empty or non-existing target " +
```
Comment (fred-db) [inline spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowIdCloneSuite.scala:95]:
```suggestion
  test("clone from empty source into an empty or non-existing target " +
```

**Review by johanl-db — APPROVED** [review]
